src/lib/utilities/is-promise.ts

```diff
@@ -0,0 +1,12 @@
+export function isPromise(obj: any): obj is Promise<unknown> {
```

Comment:
I know we talked about this a bit, but this function and its test file is what's failing the build. I wonder if we _need_ this functionality or if it's a nice to have that's adding complexity?

Do we _need_ it?